cairo-oracle-hint-processor/src/lib.rs

```diff
@@ -1,284 +1,4 @@
-    Receiving(Value),
-}
-
-/// HintProcessor for Cairo 1 compiler hints.
-pub struct RpcHintProcessor<'a> {
-    path: Vec<PathElement>,
-    state: OracleState,
-    inner_processor: CairoHintProcessor<'a>,
-    server: Option<String>,
-    configuration: &'a Configuration,
-}
-
-impl<'a> RpcHintProcessor<'a> {
-    pub fn new(
-        inner_processor: CairoHintProcessor<'a>,
-        server: &Option<String>,
-        configuration: &'a Configuration,
-    ) -> Self {
-        Self {
-            state: OracleState::Sending(Value::Null),
-            path: Default::default(),
-            inner_processor,
-            server: server.clone(),
-            configuration,
-        }
-    }
-
-    /// Executes a cheatcode.
-    fn execute_cheatcode(
-        &mut self,
-        selector: &BigIntAsHex,
-        [input_start, input_end]: [&ResOperand; 2],
-        [output_start, output_end]: [&CellRef; 2],
-        vm: &mut VirtualMachine,
-        exec_scopes: &mut ExecutionScopes,
-    ) -> Result<(), HintError> {
-        // Parse the selector.
-        let selector = &selector.value.to_bytes_be().1;
-        let selector = std::str::from_utf8(selector).map_err(|_| {
-            HintError::CustomHint(Box::from("failed to parse selector".to_string()))
-        })?;
-
-        // Extract the inputs.
-        let input_start = extract_relocatable(vm, input_start)?;
-        let input_end = extract_relocatable(vm, input_end)?;
-        let inputs = vm_get_range(vm, input_start, input_end)?;
-
-        let mut res_segment = MemBuffer::new_segment(vm);
-        let res_segment_start = res_segment.ptr;
-
-        let Some(configuration) = self.configuration.services.iter().find_map(|(_, methods)| {
-            methods.methods.get(selector)
-        }) else {
-            return Err(HintError::CustomHint(Box::from(format!(
-                "Unknown cheatcode selector: {selector}"
-            ))));
-        };
-
-        // println!("Configuration {:#?}", self.configuration);
-
-        let data = deserialize_cairo_serde(
-            self.configuration,
-            &configuration.input,
-            &mut inputs.as_ref(),
-        );
-        println!("let the oracle decide... Inputs: {data:?}");
-
-        let client = reqwest::blocking::Client::new();
-        let resp = client
-            .post(self.server.as_ref().unwrap())
-            .json(&data)
-            .send()
-            .unwrap()
-            .json::<Value>()
-            .unwrap();
-
-        let output = &resp["result"];
-        let data = serialize_cairo_serde(self.configuration, &configuration.output, output);
-        println!("Output: {output}");
-        res_segment.write_data(data.iter())?;
-
-        let res_segment_end = res_segment.ptr;
-        insert_value_to_cellref!(vm, output_start, res_segment_start)?;
-        insert_value_to_cellref!(vm, output_end, res_segment_end)?;
-        Ok(())
-    }
+pub mod rpc_1_hint_processor;
+pub mod rpc_hint_processor;
```

Comment:
why is this module needed?

Reply:
I thought to create two rpc hint processors: one that takes `Cairo1HintProcessor` and one that takes `CairoHintProcessor`

Reply:
We can ignore Cairo0 - it had its own hints system, and our protocol buffers output is incompatible with the old syntax.